### PR TITLE
[.NET] Turkish Numbers (Cardinal, Ordinal) Support

### PR DESCRIPTION
**Pull request by LionbridgeCS**
Number (Cardinal)
    Passed: 121
    Skipped: 5
Skips : 
Decimal position extraction : "192.168.1.2" 3 numbers extracted instead of 4 and "1.234.567,89"  2  numbers extracted instead of 1. 
Fraction resolution:  0,5 actual, expected 0.5; 0,75 actual, expected 0.75; 0,5 actual, 1.5 expected
**********************************************************************************************************
OrdinalModel
    Passed: 15
**********************************************************************************************************
Percent to be included in a separate PR

### CODE REVIEW

File: Patterns/Turkish/Turkish-Numbers.yaml [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+  references: [BillionsNumberIntegerRegex, MillionsNumberIntegerRegex, ThousandsNumberIntegerRegex, HundredsNumberIntegerRegex, TensNumberIntegerRegex, OneToNineDenominatorSuffixRegex, TenToHundredDenominatorSuffixRegex, HundredToThousandDenominatorSuffixRegex, ThousandToMillionDenominatorSuffixRegex, MillionToBillionDenominatorSuffixRegex, BillionsDenominatorSuffixRegex]
+AllDenominatorSuffixRegex: !nestedRegex
+  def: ({OneToNineDenominatorSuffixRegex}|{TenToHundredDenominatorSuffixRegex}|{HundredToThousandDenominatorSuffixRegex}|{ThousandToMillionDenominatorSuffixRegex}|{MillionToBillionDenominatorSuffixRegex}|{BillionToTrillionDenominatorSuffixRegex}|{RoundNumberDenominatorSuffixRegex})
+  references: [OneToNineDenominatorSuffixRegex, TenToHundredDenominatorSuffixRegex, HundredToThousandDenominatorSuffixRegex, ThousandToMillionDenominatorSuffixRegex, MillionToBillionDenominatorSuffixRegex, BillionToTrillionDenominatorSuffixRegex, RoundNumberDenominatorSuffixRegex ]
+FractionPrepositionRegex: !nestedRegex
+  def: (?<=\b)(?<numerator>({AllIntRegex})|((?<![\.,])\d+))\s+(bölü)\s+(?<denominator>({AllIntRegex})|(\d+)(?![\.,]))(?=\b)
```
Comment (tellarin) [inline Patterns/Turkish/Turkish-Numbers.yaml]: If Turkish always uses "," as decimal separator and "." for thousand separator, all occurrences of [\.,] should probably be changed to the decimal separator only "[\.,]" -> ",".

File: Patterns/Turkish/Turkish-Numbers.yaml [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+  references: [ AllFloatRegex ]
+#Percentage Regex
+NumberWithPrefixPercentage: !nestedRegex
+  def: (%|(eksi\s)?yüzde\s)({BaseNumbers.NumberReplaceToken}|{AllIntRegex})
+  references: [ BaseNumbers.NumberReplaceToken, AllIntRegex ]
+NumberWithSuffixPercentage: !nestedRegex
+  def: (((({TensNumberIntegerRegex}\s)?(birin|ikinin|üçün|dördün|beşin|altının|yedinin|sekizin|dokuzun)|onun|yirminin|otuzun|kırkın|ellinin|altmışın|yetmişin|seksenin|doksanın)\s(yüzdesi))|(\d*(1'in|2'nin|3'ün|4'ün|5'in|6'nın|7'nin|8'in|9'un|10'un|20'nin|30'un|40'ın|50'nin|60'ın|70'in|80'in|90'ın)\s(yüzdesi)))
+  references: [ TensNumberIntegerRegex ]
+FractionNumberWithSuffixPercentage: !simpleRegex
+  def: (\d+)\s+\/(\d+)?((1|5|8|70|80|\.?000)'i|(2|7|20|50)'si|(3|4|100)'ü|6'sı|(9|10|30|\.?000\.?000|\.?000\.?000\.?000\.?000)'u|(40|60|90|\.?000\.?000\.?000)'ı)
```
Comment (tellarin) [inline Patterns/Turkish/Turkish-Numbers.yaml]: Why are these needed? "\.?000\.?000|\.?000\.?000\.?000\.?000"

File: Patterns/Turkish/Turkish-Numbers.yaml (hunk 1 above)
Comment (LionbridgeCS) [reply Patterns/Turkish/Turkish-Numbers.yaml]: noted, we will try this change.

File: Patterns/Turkish/Turkish-Numbers.yaml (hunk 2 above)
Comment (LionbridgeCS) [reply Patterns/Turkish/Turkish-Numbers.yaml]: In Turkish, numbers get suffixes and it depends on the numbers so if it ends with thousands (.000), it get a certain suffix etc cases like "1 bölü 1000'i" meaning "one over 1000". 
This is solved elsewhere now.

File: Patterns/Turkish/Turkish-Numbers.yaml [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+RoundNumberIntegerRegex: !simpleRegex
+  def: (yüz|bin|milyon|milyar|trilyon)
+ZeroToNineIntegerRegex: !simpleRegex   # 0,1,2,...,9
+  def: (sıfır|bir|iki|üç|dört|beş|altı|yedi|sekiz|dokuz)
+OneToNineIntegerRegex: !simpleRegex   # 1,2,...,9
+  def: (bir|iki|üç|dört|beş|altı|yedi|sekiz|dokuz)
+TwoToNineIntegerRegex: !simpleRegex   # 2,...,9
+  def: (iki|üç|dört|beş|altı|yedi|sekiz|dokuz)
+NegativeNumberTermsRegex: !simpleRegex
+  def: ((eksi|negatif)\s+)
```
[the remark this answers is not in the document]
Comment (LionbridgeCS) [reply Patterns/Turkish/Turkish-Numbers.yaml]: Will be updated :)

[the remark this answers is not in the document]
Comment (LionbridgeCS) [reply Patterns/Turkish/Turkish-Numbers.yaml]: will be updated.

**Comment by tellarin** [comment]
@LionbridgeCS, can you create a PR for the current Numbers and Ordinal that enables only the Ordinal model and it's tests? All of those pass, so we can merge it than come back to Numbers and Percent.
@STayinloves, could you help them debug the Percent cases?

**Comment by LionbridgeCS** [comment]
@tellarin, yes, we ill update the PR with to include what passes except Percent.

**Comment by tellarin** [comment]
@LionbridgeCS please remove the 'Draft' mark if ready for considering merge.

**Comment by LionbridgeCS** [comment]
@tellarin only 5 cases are skipped for Cardinal. If after review you see this fit to be merged; we can include fixes in another PR, and include percent in a separate PR

**Comment by LionbridgeCS** [comment]
@STayinloves this PR is ready for review as only 5 cases are skipped for Cardinal. If after review you see this fit to be merged; we can include fixes in another PR, and include percent in a separate PR; because other PRs in the pipeline are dependent on this one.

**Comment by LionbridgeCS** [comment]
@STayinloves the changes are handled and test cases are added.

**Comment by hazemelh** [comment]
@STayinloves Could we merge this PR please? The team is saying they need this to be merged to make other PRs.
